flaskbb/user/views.py

```diff
@@ -179,7 +182,34 @@ def render(self):
     def redirect(self):
         return redirect(url_for("user.change_user_details"))
 
+class UserUploads(MethodView):
+    decorators = [login_required]
+
+    def get(self):
+        files = UploadedFile.query.filter_by(user_id=current_user.id).all()
+        return render_template("user/user_uploads.html",files=files)
+
+class DeleteFile(MethodView):
+    decorators = [login_required]
+
+    def post(self):
+        file_id = request.args.get("file_id",0,type=int)
+        file_record = UploadedFile.query.filter_by(id=file_id).first_or_404()
+        file_owner = User.query.filter_by(id=file_record.user_id).first_or_404()
+        
+        if not (file_owner.user_id == current_user.id or self.get_permissions()['admin']):
```

Comment:
```suggestion
        if not (file_owner.user_id == current_user.id or has_permission(current_user, "admin"):
```
Без понятия что такое **self**.get_permissions и откуда оно берет юзера. Мой вариант мне кажется более явный

Reply:
Только надо еще импортировать has_permission, да